source/robot_model/tests/testRobotModel.cpp

```diff
@@ -1,8 +1,89 @@
 #include "robot_model/Model.hpp"
+
 #include <gtest/gtest.h>
 
+#include "robot_model/Exceptions/InvalidJointStateSizeException.hpp"
+#include "robot_model/Exceptions/FrameNotFoundException.hpp"
+
 using namespace RobotModel;
 
+class RobotModelTest : public testing::Test {
+protected:
+  void SetUp() override {
+    robot_name = "franka";
+    urdf = "/tmp/panda_arm.urdf";
+    franka = Model(robot_name, urdf);
+    joint_state = StateRepresentation::JointState(robot_name, 7);
+  }
+
+  Model empty = Model();
+  Model franka;
+  std::string robot_name;
+  std::string urdf;
+
+  StateRepresentation::JointState joint_state;
+};
+
+// test setters and getters
+TEST_F(RobotModelTest, TestName) {
+  empty.set_robot_name(robot_name);
+  EXPECT_EQ(empty.get_robot_name(), robot_name);
+}
+
+TEST_F(RobotModelTest, TestUrdfPath) {
+  empty.set_urdf_path(urdf);
+  EXPECT_EQ(empty.get_urdf_path(), urdf);
+}
+
+TEST_F(RobotModelTest, TestConstructor) {
+  Model tmp(robot_name, urdf);
+  EXPECT_NO_THROW(franka = tmp);
+}
+
+TEST_F(RobotModelTest, TestNbJoint) {
+  EXPECT_EQ(franka.get_nb_joints(), 7);
+}
+
+TEST_F(RobotModelTest, TestForwardGeometryJointStateSize) {
+  StateRepresentation::JointState dummy = StateRepresentation::JointState(robot_name, 6);
+  EXPECT_THROW(franka.forward_geometry(dummy), Exceptions::InvalidJointStateSizeException);
+}
+
+TEST_F(RobotModelTest, TestForwardGeometry) {
+  EXPECT_EQ(franka.forward_geometry(joint_state).get_position(),
+            franka.forward_geometry(joint_state, "panda_link8").get_position());
+}
+
+TEST_F(RobotModelTest, TestForwardGeometryInvalidFrameName) {
+  EXPECT_THROW(franka.forward_geometry(joint_state, "panda_link99"), Exceptions::FrameNotFoundException);
+}
+
+// this fails, we should make forward geometry with frame ids either private or test for frame id existence
+//TEST_F(RobotModelTest, TestForwardGeometryInvalidFrameID) {
```

Comment:
I recommend you use the DISABLED_ macro instead of commenting out this test case. It can be used to [temporarily skip a test](https://github.com/google/googletest/blob/master/docs/advanced.md#temporarily-disabling-tests).

The usage in this case it would be:
`TEST_F(DISABLED_RobotModelTest, TestForwardGeometryInvalidFrameID) { ...`

A FIXME comment will help as well to know that this requires fixing.

Reply:
@buschbapti Would you rather have this method be private so it cannot be called from the outside with a wrong frame id, or should I make a check if the frame id is valid? in Model.cpp

Reply:
I would leave the possibility to call this function from the outside though. I can see some usage.